sa/sysvars.go

```diff
@@ -0,0 +1,219 @@
+package sa
+
+import (
+	"fmt"
+	"regexp"
+)
+
+var (
+	checkStringQuoteRE = regexp.MustCompile(`^'[0-9A-Za-z_\-=:]+'$`)
+	checkIntRE         = regexp.MustCompile(`^\d+$`)
+	checkImproperIntRE = regexp.MustCompile(`^'\d+'$`)
+	checkNumericRE     = regexp.MustCompile(`^\d+(\.\d+)?$`)
+	checkBooleanRE     = regexp.MustCompile(`^([0-1])|(?i)(true|false)|(?i)(on|off)`)
+)
+
+// checkMariaDBSystemVariables validates a MariaDB config passed in via SA
+// setDefault or DSN. This manually curated list of system variables was
+// partially generated by a tool in issue #6687. An overview of the validations
+// performed are:
+//
+//   - Correct quoting for strings and string enums prevent future
+//     problems such as PR #6683 from occurring.
+//
+//   - Regex validation is performed for the various booleans, floats, integers, and strings.
+//
+// Only session scoped variables should be included. A session variable is one
+// that affects the current session only. Passing a session variable that only
+// solely in the global scope causes database connection error 1045.
```

Comment:
```suggestion
// that affects the current session only. Passing a session variable that only
// works in the global scope causes database connection error 1045.
```

Not sure what the intended verbiage was but here's a guess!